File: frontend/src/app/models/Complaint.ts
import { Employee } from './Employee';
import { ProjectManager } from './ProjectManager';

export class Complaint {
  idComplaint: number =0;
  title: string = '';
  description: string = '';
  issueDate: Date = new Date();
  status: boolean = false;
  sender: Employee = new Employee();
  receiver: ProjectManager = new ProjectManager();
}

File: backend/src/main/java/tn/company/taskManagement/entities/Complaint.java
package tn.company.taskManagement.entities;

import jakarta.persistence.*;
import lombok.*;

import java.util.Date;

@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Complaint {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int idComplaint;
    private String title;
    private String description;

    @Temporal(TemporalType.DATE)
    private Date issueDate;

    private boolean status;

    @ManyToOne
    @JoinColumn(name = "employee_id")
    private Employee sender;

    @ManyToOne
    @JoinColumn(name = "project_manager_id")
    private ProjectManager receiver;
}
